Draft Pull Request for Author Toolkit

jbrock1999h opened the pull request:
This pull request adds codespace container files that will create the platform that authors use to easily edit their textbooks. The end goal is to have a button within the published textbook that links directly back to this codespace, allowing the textbook to be quickly and easily edited. So far, we have a codespace set up, but we do not yet have it as user friendly as we want it to be because it requires more git CLI manipulation than we want to get started.

This is an assignment for Josh Harvey and Daniel Short.

jbrock1999h committed:
Add: Codespace container files

jbrock1999h committed:
publication

jbrock1999h committed:
Add: Codepsaces container files
Modify: Added container post-creation commands for git usability and pretext project pre-building
Modify: Added new paths to ignore for source control

jbrock1999h committed:
Modify: wording

bjones1 commented on .devcontainer/devcontainer-20230420-004357.json:
Remove this file from the commit -- it was auto-edited, so we no longer care about this stuff.

bjones1 commented on .gitignore-20230420-004357:
Remove this file also.

bjones1 commented on project-20230420-004357.ptx:
Remove...

jbrock1999h committed:
Delete devcontainer-20230420-004357.json

jbrock1999h committed:
Delete .gitignore-20230420-004357

jbrock1999h committed:
Delete project-20230420-004357.ptx

jbrock1999h reviewed (commented):
The requested files have been removed from the pull request

jbrock1999h committed:
Remove: Extraneous files
Modify: Build files; we want the pretext commands and git commands to run on startup, and this is a test for that.

jbrock1999h committed:
Initial commit

jbrock1999h committed:
Modify: Test

jbrock1999h committed:
Modify docker build for hub include

jbrock1999h committed:
Modify: Startup commands for testing

jbrock1999h committed:
Modify: Test

jbrock1999h committed:
Modify: devcontainer.json to add live share

jbrock1999h committed:
Modify: PostStartCommands to view book on startup and test git

jbrock1999h committed:
Modify: Devcontainer for bash instruction change

jbrock1999h committed:
Modify: post start commands to test web building

jbrock1999h committed:
Modify: Testing for pretext view web automation

jbrock1999h committed:
Modify: More testing for prext view page opening automation

jbrock1999h committed:
More Testing

bjones1 commented on .devcontainer/postStartCommand.sh:
```diff
@@ -0,0 +1,14 @@
+#pretext generate
+pretext view web &
+# The command pretext view web starts a local server which creates a process that does not end until the user manually ends it. This means that any command that comes after pretext view web will not run unless the local server is manually closed.
+xdg-open http://localhost:8000-.preview.app.github.dev
```
I don't understand how this command (open a specific URL) relates to the previous comment (which says this command won't run?).

bjones1 commented on .devcontainer/postCreateCommand.sh:
```diff
@@ -0,0 +1,13 @@
+# Create a new PreTeXt project.
+pretext new -d .
+
+# For the Digital System Design book...
```
You can remove all this.

bjones1 commented on .devcontainer/docker-build.sh:
```diff
@@ -0,0 +1,16 @@
+
+apt install -y hub
+
+#apt install -y xdg-utils
```
You seem to run this later, so uncomment it? (But I don't understand why running it is necessary...)

bjones1 commented on .devcontainer/postStartCommand.sh:
```diff
@@ -0,0 +1,14 @@
+#git reset --hard origin/main
+
+# Build pretext project
+#pretext build
```
The pretext generate and pretext build should happen in the Docker container build, not here.

bjones1 reviewed (commented):
Here are fixes/updates for your next PR; this is fine for the draft PR.
